### ISSUE TITLE
example application using ldap for authentication

### ISSUE THREAD

**Issue by mnlbox** [hasura/graphql-engine#1428]
Hi guys,
Can you guide me how to implement Hasura with a LDAP active directory authentication?
I think it's useful if we can provide some of this frequently asked feature as predefine docker-compose file.
For example with [OpenLDAP](https://github.com/osixia/docker-openldap) and [PHP LDAP Admin](https://github.com/osixia/docker-phpLDAPadmin).

**Comment by tirumaraiselvan** [comment]
Hi @mnlbox 

Since Hasura does not package any auth systems by default, we can't put a LDAP server in official docker installations.

We would love the community to write a simple boilerplate which connects to a LDAP server and authenticates the user. The basic boilerplate should include:

1) Script/docs to setup a LDAP server which manages username, password, allowed-roles.
2) A LDAP web service which is a LDAP client that reads data from, say, Authorization header which contains base64 encoded value of "user:password:role". It passes this info to LDAP server and receives True or False or allowed-roles. The web service returns appropriate session variables as response.

3) Instructions to deploy this LDAP web service.

**Comment by mnlbox** [comment]
@tirumaraiselvan Maybe we can do this with Keycloak and [keycloak-hasura-connector](https://github.com/httpsOmkar/keycloak-hasura-connector/) :thinking: 

Also related: #1779

**Comment by lobosan** [comment]
Yes actually, that's why I needed support for keyclock because I need to use active directory for internal users and gmail for external ones. A video tutorial would be amazing

**Comment by davie-robertson** [comment]
@lobosan I have the same requirement, I think we can also do the gmail login with Auth0 and they also have an LDAP connector for AD.  This this isn't all on premise, which I'm guessing you need.

**Comment by lobosan** [comment]
@davie-robertson yes you right
